src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+    }
+
+    /**
+     * Sets a user id.
+     * @param userId reference new user id for the proposition
+     */
+    public void setUserId(final int userId) {
+        this.set("user_id", userId);
+    }
+
+    /**
+     * Sets a challenge id.
+     * @param challengeId reference new id challenge for the proposition
+     */
+    public void setChallengeId(final int challengeId) {
+        this.set("challenge_id", challengeId);
+    }
+
+    /**
+     * Sets a source's.
+     * @param source reference new source for the proposition
+     */
+    public void setSource(final String source) {
+        if (source == null) {
+            throw new IllegalArgumentException();
+        }
+        this.set("source", source);
+    }
+
+    /**
+     * Set isSolution value.
+     * @param isSolution reference new value
+     * for the isSolution in the proposition
+     */
+    public void setIsSolution(final boolean isSolution) {
+        this.set("isSolution", isSolution);
+    }
+
+    /**
+     * Set distance value calculated previus.
+     * @param distance reference new distance in the proposition.
+     */
+    public void setDistance(final int distance) {
+        this.set("distance", distance);
+    }
+
+    /**
+     * Set cantTestPassed value.
+     * @param testPassed reference the new amount of tests passed
+     */
+    public void setCantTestPassed(final int testPassed) {
+        this.set("cantTestPassed", testPassed);
+    }
+
+    /**
+     * Method for creates new proposition in the database.
+     * @param userId reference the user for the new proposition.
+     * @param challengeId reference the challenge for the new proposition.
+     * @return the proposition created in the database.
+     */
+    public static Proposition newProposition(
+        final int userId,
+        final int challengeId) {
+
+        Challenge currentChallenge = Challenge.findById(challengeId);
+        User currentUser = User.findById(userId);
+
+        if (currentChallenge == null || currentUser == null) {
+            throw new IllegalArgumentException();
+        }
+
+        Proposition newProposition = new Proposition();
+        newProposition.set("user_id", currentUser.getInteger("id"));
+        newProposition.set("challenge_id", currentChallenge.getInteger("id"));
+        newProposition.set("source", currentChallenge.get("source"));
+        newProposition.set("isSolution", false);
+        newProposition.set("distance", 0);
+        newProposition.set("cantTestPassed", 0);
+        newProposition.saveIt();
+        return newProposition;
+    }
+
+    /**
+     * Get the solutions for a challenge by a specific user..
+     * @param userId reference the id of a user
+     * @param challengeId reference the id of a challenge
+     * @return List of proposition for user in challenge
+     */
+    public static LazyList<Proposition> getChallengeSolutionsByUser(
+            final Integer userId, final Integer challengeId) {
```

Comment:
`int` should be used instead of `Integer`, when using `Integer` a `null` check should be included.